web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -900,7 +900,7 @@
                   imageTile.getImage().src = src;
                 },
                 function (r) {
-                  if (r.status === 414) {
+                  if (r.status && r.status === 414) {
```

Comment:
What does that new check achieve? It seems superfluous

Reply:
If CORS is not enabled, the response object has no status

![image](https://github.com/geonetwork/core-geonetwork/assets/1701393/f6aa0e1f-8670-46eb-a227-c78b41fc6e54)

It looks like some servers add CORS header on GET and not on HEAD. Also depending on Apache config, if there is an error, then CORS header are not always present (depends on virtual host config and the set header directive).

The CORS interceptor does not make the difference between operation so it will start using the proxy for GET if an HEAD request fails (and then GET request may also be not allowed causing subsequent issues).

Reply:
@jahow without that check, when the response has no status (as described previously by @fxprunayre ), it fails accessing `r.status`, what causes the CORS interceptor to add the server in the list to use the http proxy when accessing any url from that server. 

So if you are unlogged and try to add a layer from that server in the map viewer (Add layers > Services > WMS), you get the following error:

![map-error](https://github.com/geonetwork/core-geonetwork/assets/1695003/9ecaf17c-c67a-43c2-97f7-8a380fa8b6a8)

Due to the http proxy default policy to require the service to be defined in the metadata and analysed with the link analysis tool, for unlogged users. The service seems working fine with CORS (at least for GET operations).

Reply:
No but I meant, if r.status equals 414 then it has to be truthy; I just don't get the `r.status &&` part

Reply:
@jahow see the commment from @fxprunayre in his reply:

> If CORS is not enabled, the response object has no status

Reply:
I understand but:
* if `r.status` is undefined then `r.status === 414` will return false
* if `r.status` is undefined then `r.status && r.status === 414` will return undefined
* if `r.status` is 414 then `r.status === 414` will return true
* if `r.status` is 414 then `r.status && r.status === 414` will also return true
* if `r.status` is 200 then `r.status === 414` will return false
* if `r.status` is 200 then `r.status && r.status === 414` will also return false

I'm just not sure this change is going to do anything at all?

Reply:
@jahow you are totally right, apologies. I was testing several options and I seems I got confused with the one that worked.

I just updated the code to avoid the CORS interceptor for the HEAD request, that is working fine. Doesn't seem optimal, but at least looks working fine.